android/app/src/main/java/org/electroncash/electroncash3/ColdLoad.kt

```diff
@@ -0,0 +1,105 @@
+package org.electroncash.electroncash3
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Intent
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.ViewModel
+import com.google.zxing.integration.android.IntentIntegrator
+import kotlinx.android.synthetic.main.load.*
+import kotlinx.android.synthetic.main.main.*
+
+// This provides a dialog to allow users to input a sting, which is then broadcast
+// on the bitcoin cash network. Strings are not validated, so invalid user inputs fail
+// gracefully and nothing happens. Valid transactoin quickly show up in transactions.
+
+class ColdLoadDialog : AlertDialogFragment() {
+
+    class Model : ViewModel() {}
+
+    //    val coldloadDialog by lazy { targetFragment as ColdLoadDialog }
+    val model: Model by viewModels()
+
+
+    init {
+        if (daemonModel.wallet!!.callAttr("is_watching_only").toBoolean()) {
+            throw ToastException(R.string.this_wallet_is)
+        } else if (daemonModel.wallet!!.callAttr("get_receiving_addresses")
+                        .asList().isEmpty()) {
+            // At least one receiving address is needed to call wallet.dummy_address.
+            throw ToastException(
+                    R.string.electron_cash_is_generating_your_addresses__please_wait_)
+        }
+    }
+
+    override fun onBuildDialog(builder: AlertDialog.Builder) {
+        builder.setTitle(R.string.load_unbroadcasted)
+                .setView(R.layout.load)
+                .setNegativeButton(android.R.string.cancel, null)
+                .setNeutralButton(R.string.qr_code, null)
+                .setPositiveButton(android.R.string.ok, null)
```

Comment:
I think the "OK" button should be changed to "Send" so it's clear what it does.

Reply:
changed to r.string.sign